Build failure: pkgsMusl.systemd

### Steps To Reproduce

1. nix build .#pkgsMusl.systemd

Note: Requires a cherry-pick ([0b9653816fd556a127b0b9600f7f67cf34de40e5](https://github.com/NixOS/nixpkgs/commit/b4ac0c8d9863b80b5fa7086c7be30a975e0ff4fa)) from staging for openexr-2.5.8. (Unless commit has already reached master.)

### Build log

```
1039/1596] Compiling C object systemd-coredump.p/src_coredump_coredump.c.o
In file included from ../src/basic/macro.h:446,
                 from ../src/shared/acl-util.h:15,
                 from ../src/coredump/coredump.c:16:
../src/coredump/coredump.c: In function ‘grant_user_access’:
../src/basic/errno-util.h:22:67: warning: pointer/integer type mismatch in conditional expression
   22 | #define STRERROR_OR_EOF(errnum) ((errnum) != 0 ? STRERROR(errnum) : "Unexpected EOF")
      |                                                                   ^
../src/basic/log.h:214:86: note: in definition of macro ‘log_full_errno_zerook’
  214 |                         ? log_internal(_level, _e, PROJECT_FILE, __LINE__, __func__, __VA_ARGS__) \
      |                                                                                      ^~~~~~~~~~~
../src/basic/log.h:253:41: note: in expansion of macro ‘log_full_errno’
  253 | #define log_warning_errno(error, ...)   log_full_errno(LOG_WARNING, error, __VA_ARGS__)
      |                                         ^~~~~~~~~~~~~~
../src/coredump/coredump.c:368:24: note: in expansion of macro ‘log_warning_errno’
  368 |                 return log_warning_errno(errno_or_else(EIO),
      |                        ^~~~~~~~~~~~~~~~~
../src/coredump/coredump.c:369:82: note: in expansion of macro ‘STRERROR_OR_EOF’
  369 |                                          "Failed to pread from coredump fd: %s", STRERROR_OR_EOF(errno));
      |                                                                                  ^~~~~~~~~~~~~~~
```
[Log file](https://github.com/NixOS/nixpkgs/files/13928168/systemd-musl.txt)

### Additional context
<!--
- Ongoing bisect (builds are very slow here) => Ok, I need to redo this. Out of space in disk.
  - Good:
    - 4d2b37a84fad1091b9de401eb450aae66f1a741e [22.11]
    - 550e897a481b769bda516e4543166796fb455b7d
    - f6242f9557d49887bdd5cf881f8f9472e13d5890
    - b1ae965cda091b14fd37c86d783ae11e836d02e0
    - d54e7189734b0aca447ee8e509ac0be8a1bbd4af
    - a84061914844e5b3c05ae7d4aaa3357e404a954c
    - 9d6b7362f1dc398e26a96fd50a3d20b273a660e7
    - b1bf1517d0da7aa356ffbce2be949998d3b12fac
    - a1e50f1f34089eebd25ddf55f5fdc2e9920d58c1
    - 8c0956e103392e4b69cff475c9b673c9692ac579
    - 9c54ff2de70bc5cdbfff5542a80f2f68e4db4135
    - 018426f66f597411bbc1191a1dbb9339d6106158
    - 886dc9e5d328662a91465b4663381dfc209fb083
    - 8664c8a5f75d805eb5157bcb79c990257c4d7f46
    - cbef2c9c4cf0947b91c9021b7be59175b182c057
    - 550119586962c42f780fd4adc9b674fc84b6bd5b
  - Bad:
    - 887e96e875e933c0a5d203ffc40e202f3d16317e
    - d6c86a2a6671a4cd4293dce6e1c1b9d93d090dfc
    - b0e46d72a8858ade0e7bf801ae10275c46ee5f46
    - 8e5a5551e8eb869a8bf7e737f5e97fb5a6ad9d26
    - fa73b913d7a7b78a3625b8f1b2c84328c6751d88
    - d014d27fd639dca83fe68080612faf14b325592c
    - 011e924109e77bb5e958b2a1adf230c34d5d83ca
    - 6008246790d4b3eb27c02d3a9b9424b91f55a8b4

-->

* Git bisect: 6008246790d4b3eb27c02d3a9b9424b91f55a8b4
  - systemd: disable NSCD when DNSSEC validation is disabled in timesyncd

This patch breaks systemd for musl. And builds fine when wrapped with:

> ] ++ lib.optional (!stdenv.hostPlatform.isMusl) [
>   ./0020-timesyncd-disable-NSCD-when-DNSSEC-validation-is-dis.patch
> ]

So I ***NAIVELY*** suggest to just disable it for musl?

Proposed PR: #281068

### Notify maintainers

@flokli @kloenk @yu-re-ka @risicle

--- a/pkgs/os-specific/linux/systemd/default.nix
+++ b/pkgs/os-specific/linux/systemd/default.nix
@@ -208,6 +208,7 @@ stdenv.mkDerivation (finalAttrs: {
     ./0017-core-don-t-taint-on-unmerged-usr.patch
     ./0018-tpm2_context_init-fix-driver-name-checking.patch
     ./0019-systemctl-edit-suggest-systemdctl-edit-runtime-on-sy.patch
+  ] ++ lib.optional (stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isGnu) [
     ./0020-timesyncd-disable-NSCD-when-DNSSEC-validation-is-dis.patch
   ] ++ lib.optional stdenv.hostPlatform.isMusl (
     let

